File: 2576-minimum-penalty-for-a-shop/2576-minimum-penalty-for-a-shop.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        best=cur = customers.count("N")
        res=len(customers)
        for i in range(len(customers)-1,-1,-1):
            cur+=1 if customers[i]=="Y" else -1
            if cur <=best:
                best=cur
                res=i
        return res

File: 2576-minimum-penalty-for-a-shop/2576-minimum-penalty-for-a-shop.cpp
class Solution {
public:
    int bestClosingTime(string customers) {
        int cur=0, res=0, idx=customers.length();
        for (int i = customers.length()-1; i >=0; i--) {
            cur += (customers[i] == 'Y' )? 1 : -1;
            if (cur <= res) {
                res = cur;
                idx = i;
            }
        }
        return idx;
    }
};